### PR TITLE
Flowcell backend

### PR DESCRIPTION
**Pull request by mayabrandi**
Adding flowcell data to the database. 

Tests will come.

### CODE REVIEW

File: vogue/load/flowcell.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from vogue.build.flowcell import build_run
```
Comment (moonso) [inline vogue/load/flowcell.py]:
```suggestion
import logging
from vogue.build.flowcell import build_run
```

File: vogue/load/flowcell.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from vogue.constants.constants import RUN_TYPES, INSTRUMENTS
+
```
Comment (moonso) [inline vogue/load/flowcell.py]:
```suggestion
LOG = logging.getLogger(__name__)
```

File: vogue/load/flowcell.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+def load_one(adapter, run):
+    """Function to load one lims flowcell into the database"""
+
+    run_id = run.udf.get('Run ID')
+    if not run_id:
+        return
```
Comment (moonso) [inline vogue/load/flowcell.py]:
```suggestion
        LOG.warning("Run ID is missing from udf")
        return
```

File: vogue/load/flowcell.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    date, instrument = run_id.split('_')[0:2]
+    insrument_name =  INSTRUMENTS.get(instrument)
```
Comment (moonso) [inline vogue/load/flowcell.py]:
```suggestion
    instrument_name =  INSTRUMENTS.get(instrument)
```

File: vogue/load/flowcell.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    if not insrument_name:
```
Comment (moonso) [inline vogue/load/flowcell.py]:
```suggestion
    if not instrument_name:
        LOG.warning("Instrument name is missing in udf")
```

File: vogue/load/flowcell.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        return
+    mongo_run = build_run(run=run, instrument = insrument_name, date=date)
+    if mongo_run.get('_id'):
```
Comment (moonso) [inline vogue/load/flowcell.py]: What if '_id' is not there??

File: vogue/parse/build/flowcell.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
+from vogue.constants.constants import LANE_UDFS
+import numpy as np
+import datetime as dt
+import math
+
+def filter_none(mongo_dict):
+    """Function to filter out Nones and NaN from a dict."""
+    for key in list(mongo_dict.keys()):
+        val = mongo_dict[key]
+        if isinstance(val,(dict, str, dt.datetime)):
```
Comment (moonso) [inline vogue/parse/build/flowcell.py]: You are checking for None and Nan so this if could be skipped, right?

File: vogue/load/flowcell.py (hunk 6 above)
Comment (mayabrandi) [reply vogue/load/flowcell.py]: It is taken care of in the load..

File: vogue/parse/build/flowcell.py (hunk 7 above)
Comment (mayabrandi) [reply vogue/parse/build/flowcell.py]: math.isnan(val) only accepts ints and floats. I didnt know how to hande it in another way. Happy for suggestions.

File: vogue/load/flowcell.py (hunk 5 above)
Comment (mayabrandi) [reply vogue/load/flowcell.py]: 👍

File: vogue/load/flowcell.py (hunk 4 above)
Comment (mayabrandi) [reply vogue/load/flowcell.py]: 👍

File: vogue/load/flowcell.py (hunk 3 above)
Comment (mayabrandi) [reply vogue/load/flowcell.py]: 👍

File: vogue/load/flowcell.py (hunk 2 above)
Comment (mayabrandi) [reply vogue/load/flowcell.py]: 👍

**Comment by mayabrandi** [comment]
Tests added @moonso

File: vogue/parse/build/flowcell.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+            continue
+        if val is None or math.isnan(val):
+            mongo_dict.pop(key)
+    return mongo_dict
+
+
+def run_data(run):
```
Comment (northwestwitch) [inline vogue/parse/build/flowcell.py]: Could this be tested?

File: vogue/commands/load/flowcell.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+LOG = logging.getLogger(__name__)
+
+@click.command("flowcell", short_help = "load flowcell into db.")
+@click.option('-r', '--run-id', help = 'Lims process id for the run. Eg: 24-100451')
+@click.option('-a', '--all-runs', is_flag = True, help = 'Loads all lims flowcells ids')
+@click.option('--dry', is_flag = True, help = 'Load from flowcell or not. (dry-run)')
+
+@with_appcontext
+def flowcell(run_id, all_runs, dry):
```
Comment (northwestwitch) [inline vogue/commands/load/flowcell.py]: This should definitely be tested I think!

File: vogue/parse/build/flowcell.py (hunk 8 above)
Comment (mayabrandi) [reply vogue/parse/build/flowcell.py]: It is tested indirectly, as part of the tests of build_run. But maybe it is better to separate the tests for build_run and run_data?

Comment (northwestwitch) [reply vogue/parse/build/flowcell.py]: If it's tested, even if indirectly, it's fine!

File: vogue/commands/load/flowcell.py (hunk 9 above)
Comment (mayabrandi) [reply vogue/commands/load/flowcell.py]: 👍 Will do!

Comment (mayabrandi) [reply vogue/commands/load/flowcell.py]: Okej! I added tre tests for this function now! @northwestwitch

File: vogue/load/flowcell.py (hunk 6 above)
Comment (mayabrandi) [reply vogue/load/flowcell.py]: 👍

File: vogue/load/flowcell.py (hunk 3 above)
Comment (mayabrandi) [reply vogue/load/flowcell.py]: This one is ok now I think..? I remover the "if not run_id:" and tested it. It works fine. Don't remember if there was anything else you wanted me to change before approving..?

**Pull request merged**